File: Tutoriels/Cpp/Make/code/Librairie_Dynamique_Creation/src/manager/GMath.cpp
//===============================================
#include "GMath.h"
//===============================================
GMath* GMath::m_instance = 0;
//===============================================
GMath::GMath() {

}
//===============================================
GMath::~GMath() {

}
//===============================================
GMath* GMath::Instance() {
    if(m_instance == 0) {
        m_instance = new GMath;
    }
    
    return m_instance;
}
//===============================================
double GMath::add(const double& a, const double& b) {
    double lData = a + b;
    return lData;
}
//===============================================
double GMath::subtract(const double& a, const double& b) {
    double lData = a - b;
    return lData;
}
//===============================================
double GMath::multiply(const double& a, const double& b) {
    double lData = a * b;
    return lData;
}
//===============================================
double GMath::divide(const double& a, const double& b) {
    double lData = a / b;
    return lData;
}
//===============================================
double GMath::quotient(const double& a, const double& b) {
    double lData = floor(divide(a, b));
    return lData;
}
//===============================================
double GMath::modulo(const int& a, const int& b) {
    int lData = a % b;
    return double(lData);
}
//===============================================

File: Tutoriels/CSharp/CSC/Module_Creation/src/manager/GMath.cs
//===============================================
using System;
//===============================================
public class GMath {
    //===============================================
    private static GMath m_instance = null;
    //===============================================
    private GMath() {
        
    }
    //===============================================
    public static GMath Instance() {
        if(m_instance == null) {
            m_instance = new GMath();
        }
        return m_instance;
    }
    //===============================================
    public double Add(double a, double b) {
        double lData = a + b;
        return lData;
    }
    //===============================================
    public double Sub(double a, double b) {
        double lData = a - b;
        return lData;
    }
    //===============================================
    public double Mul(double a, double b) {
        double lData = a * b;
        return lData;
    }
    //===============================================
    public double Div(double a, double b) {
        double lData = a / b;
        return lData;
    }
    //===============================================
}
//===============================================
